### PR TITLE
Add proper termination to healClient tests

### PR DESCRIPTION
**Pull request by lvfxx**

### CODE REVIEW

**Comment by lvfxx** [comment]
Except `TestNewClient_MissingConnectionsOnInit` test. See #160 about it

File: pkg/networkservice/common/heal/client.go [hunk 1]
```diff
@@ -163,12 +163,11 @@ func (f *healClient) Request(ctx context.Context, request *networkservice.Networ
 	// Set its connection to the returned connection we received
 	req.Connection = rv
 
-	// TODO handle deadline err
-	deadline, _ := ctx.Deadline()
-	duration := time.Until(deadline)
+	// TODO define proper constant timeout
+	timeout := time.Minute
```
Comment (edwarnicke) [inline pkg/networkservice/common/heal/client.go]: Why 1 minute here?

Comment (denis-tingajkin) [inline pkg/networkservice/common/heal/client.go]:
```suggestion
	// TODO handle deadline err
	deadline, _ := ctx.Deadline()
	duration := time.Until(deadline)
	f.updateExecutor.AsyncExec(func() {
		f.requestors[req.GetConnection().GetId()] = func() {
			        // TODO rename timeCtx
		                 var  timeCtx context.Context
		                 var  cancelFunc func()
		                 if duration > 0 {
		                                 timeCtx, cancelFunc = context.WithTimeout(f.chainContext, duration)
		                 } else {
		                                 timeCtx, cancelFunc = context.WithCancel(f.chainContext)
		                 }
```

Comment (lvfxx) [reply pkg/networkservice/common/heal/client.go]:
```suggestion
	deadline, hasDeadline  := ctx.Deadline()
	f.updateExecutor.AsyncExec(func() {
		f.requestors[req.GetConnection().GetId()] = func() {
			        // TODO rename timeCtx
		                 var  timeCtx context.Context
		                 var  cancelFunc func()
		                 if hasDeadline {
		                                 timeCtx, cancelFunc = context.WithDeadline(f.chainContext, deadline)
		                 } else {
		                                 timeCtx, cancelFunc = context.WithCancel(f.chainContext)
		                 }
```
I like your variant, but i wonder why don't just use `context.WithDeadline()`? When use `context.WithDuration()` context for `f.onHeal` may remain valid after request's context deadline expiration due to possibly defferred execution of `f.updateExecutor.AsyncExec(func() {...})`. But with `context.WithDeadline()` it's guaranteed that both contexts have the same cancellation moment.

Comment (lvfxx) [reply pkg/networkservice/common/heal/client.go]: @edwarnicke any thoughts?

Comment (edwarnicke) [reply pkg/networkservice/common/heal/client.go]: @lvfxx Excellent question.  Heal *intrinsically* may happen long after the deadline of the original request that created the connection.  I may have a request that comes in with 10 seconds till its deadline, resulting in a connection with an ExpireTime 1 day in the future.  12 hours in, healing is triggered.  We are basically using the initial connection to *hint* at a reasonable deadline (the same duration in the future from when we start healing)... not a source of an absolute deadline.

Comment (lvfxx) [reply pkg/networkservice/common/heal/client.go]: Ah, got it, thanks :)

Comment (lvfxx) [reply pkg/networkservice/common/heal/client.go]: I think i should remove constant timeout at all...

**Pull request merged**